### PR TITLE
Improves the PanelFrontend docs

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
I was trying out pip installing the git repository and working with the PanelFrontend and could see the docstrings where not helpful enough now that Its been a while since I worked with the PanelFrontend. There where two issues

- Not enough info
- Not rendering well as a tooltip in VS Code.

This PR fixes it.

### CODE REVIEW

File: src/lightning_app/frontend/panel/panel_frontend.py [hunk 1]
```diff
@@ -27,25 +27,40 @@ def has_panel_autoreload() -> bool:
...
+        The file or function must contain your Panel code.
+        The function can optionally accept an `AppStateWatcher` argument.
+
+    Raises
+    ------
+    TypeError : Raised if the `entry_point`provided is a class method
+
+    Example
+    -------
+
```
Comment (awaelchli) [inline src/lightning_app/frontend/panel/panel_frontend.py:46]: We use this other format (I think it's called Google style):
```suggestion
    Args:
        entry_point: The path to a .py or .ipynb file, or a pure function. The file or function must contain your Panel code.
            The function can optionally accept an `AppStateWatcher` argument.

    Raises:
        TypeError: Raised if the `entry_point`provided is a class method

    Example:

```

File: src/lightning_app/frontend/panel/panel_frontend.py [hunk 2]
```diff
@@ -69,20 +84,15 @@ def configure_layout(self):
...
-        entry_point: A pure function or the path to a .py or .ipynb file.
-        The function must be a pure function that contains your Panel code.
-        The function can optionally accept an `AppStateWatcher` argument.
+    Start the Lightning server with `lightning run app app_basic.py`.
 
-    Raises:
-        TypeError: Raised if the entry_point is a class method
+    For development you can get Panel autoreload by setting the `PANEL_AUTORELOAD`
+    environment variable to 'yes', i.e. run
+    `PANEL_AUTORELOAD=yes lightning run app app_basic.py`
```
Comment (awaelchli) [inline src/lightning_app/frontend/panel/panel_frontend.py:91]: Note: single ticks make it look something like italic, double ticks will make it an inline code piece.

**Review by awaelchli — APPROVED** [review]

File: src/lightning_app/frontend/panel/panel_frontend.py [hunk 3]
```diff
@@ -69,20 +80,15 @@ def configure_layout(self):
...
     """
 
     @requires("panel")
-    def __init__(self, entry_point: Callable | str):
+    def __init__(self, entry_point: str | Callable):
```
Comment (Borda) [inline src/lightning_app/frontend/panel/panel_frontend.py:91]:
```suggestion
    def __init__(self, entry_point: Union[str, Callable]):
```

File: src/lightning_app/frontend/panel/panel_frontend.py [hunk 4]
```diff
@@ -27,25 +27,36 @@ def has_panel_autoreload() -> bool:
...
 
-    To use this frontend, you must first install the `panel` package:
+    Reference: https://lightning.ai/lightning-docs/workflows/add_web_ui/panel/
+
+    Args:
+        entry_point: The path to a .py or .ipynb file, or a pure function. The file or function must contain your Panel code.
+            The function can optionally accept an `AppStateWatcher` argument.
+
+    Raises:
+        TypeError: Raised if the `entry_point`provided is a class method
```
Comment (Borda) [inline src/lightning_app/frontend/panel/panel_frontend.py:39]:
```suggestion
        TypeError: Raised if the ``entry_point`` provided is a class method
```

File: src/lightning_app/frontend/panel/panel_frontend.py [hunk 5]
```diff
@@ -27,25 +27,36 @@ def has_panel_autoreload() -> bool:
...
 class PanelFrontend(Frontend):
-    """The PanelFrontend enables you to serve Panel code as a Frontend for your LightningFlow.
+    """The `PanelFrontend` enables you to serve Panel code as a Frontend for your LightningFlow.
 
-    To use this frontend, you must first install the `panel` package:
+    Reference: https://lightning.ai/lightning-docs/workflows/add_web_ui/panel/
+
+    Args:
+        entry_point: The path to a .py or .ipynb file, or a pure function. The file or function must contain your Panel code.
+            The function can optionally accept an `AppStateWatcher` argument.
```
Comment (Borda) [inline src/lightning_app/frontend/panel/panel_frontend.py:36]:
```suggestion
            The function can optionally accept an ``AppStateWatcher`` argument.
```

File: src/lightning_app/frontend/panel/app_state_watcher.py [hunk 6]
```diff
@@ -54,10 +55,10 @@ def update(state):
 
     This would print ``The counter was updated to 2``.
 
-    The AppStateWatcher is built on top of Param which is a framework like dataclass, attrs and
+    The `AppStateWatcher` is built on top of Param which is a framework like dataclass, attrs and
```
Comment (Borda) [inline src/lightning_app/frontend/panel/app_state_watcher.py:58]:
```suggestion
    The ``AppStateWatcher`` is built on top of Param, which is a framework like dataclass, attrs and
```

File: src/lightning_app/frontend/panel/app_state_watcher.py [hunk 7]
```diff
@@ -54,10 +55,10 @@ def update(state):
...
     Pydantic which additionally provides powerful and unique features for building reactive apps.
 
-    Please note the AppStateWatcher is a singleton, i.e. only one instance is instantiated
+    Please note the `AppStateWatcher` is a singleton, i.e. only one instance is instantiated
```
Comment (Borda) [inline src/lightning_app/frontend/panel/app_state_watcher.py:61]:
```suggestion
    Please note the ``AppStateWatcher`` is a singleton, i.e., only one instance is instantiated
```

File: src/lightning_app/frontend/panel/app_state_watcher.py [hunk 8]
```diff
@@ -1,9 +1,9 @@
-"""The AppStateWatcher enables a Frontend to.
+"""The `AppStateWatcher` enables a Frontend to:
```
Comment (Borda) [inline src/lightning_app/frontend/panel/app_state_watcher.py:1]:
```suggestion
"""The ``AppStateWatcher`` enables a Frontend to:
```

File: src/lightning_app/frontend/panel/app_state_watcher.py [hunk 9]
```diff
@@ -1,9 +1,9 @@
...
 
 - subscribe to App state changes
 - to access and change the App state.
 
-This is particularly useful for the PanelFrontend but can be used by other Frontends too.
+This is particularly useful for the `PanelFrontend` but can be used by other frontends too.
```
Comment (Borda) [inline src/lightning_app/frontend/panel/app_state_watcher.py:6]:
```suggestion
This is particularly useful for the ``PanelFrontend`` but can be used by other frontends too.
```

File: src/lightning_app/frontend/panel/panel_frontend.py [hunk 10]
```diff
@@ -69,20 +80,15 @@ def configure_layout(self):
...
+    Start the Lightning server with `lightning run app app_basic.py`.
 
-    Args:
-        entry_point: A pure function or the path to a .py or .ipynb file.
-        The function must be a pure function that contains your Panel code.
-        The function can optionally accept an `AppStateWatcher` argument.
-
-    Raises:
-        TypeError: Raised if the entry_point is a class method
+    For development you can get Panel autoreload by setting the `PANEL_AUTORELOAD`
```
Comment (Borda) [inline src/lightning_app/frontend/panel/panel_frontend.py:85]:
```suggestion
    For development you can get Panel autoreload by setting the ``PANEL_AUTORELOAD``
```

File: src/lightning_app/frontend/panel/panel_frontend.py [hunk 11]
```diff
@@ -69,20 +80,15 @@ def configure_layout(self):
...
+    environment variable to 'yes', i.e. run
+    `PANEL_AUTORELOAD=yes lightning run app app_basic.py`
```
Comment (Borda) [inline src/lightning_app/frontend/panel/panel_frontend.py:87]:
```suggestion
    ``PANEL_AUTORELOAD=yes lightning run app app_basic.py``
```

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
@Felonious-Spellfire mind having a look ^^ :otter:

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by manskx — APPROVED** [review]

**Review by lantiga — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**